### PR TITLE
Termios processing was separated from tty

### PR DESCRIPTION
**Pull request by MaKToff**
Full description of the task:
https://trello.com/c/fNVYQQt3/10-tty-termios

### CODE REVIEW

**Comment by anton-bondarev** [comment]
Please extract only meaningful changes.
And fix templates arm/qemu, x86/qemu, x86/test/lang

**Comment by anton-bondarev** [comment]
@MaKToff Please first of all fix [this](https://travis-ci.org/embox/embox/jobs/213369216)
Your changes break ppty_test.c which check termios

File: src/drivers/tty/termios_ops.c [hunk 1]
```diff
@@ -116,11 +131,12 @@ int termios_handle_newline(const struct termios *t, char ch, int *is_eol) {
 	return ignore_cr;
 }
 
-int termios_handle_erase(const struct termios *t, char ch, int *raw_mode,
-	struct ring *i_ring, struct ring *i_canon_ring, size_t buflen) {
+int termios_handle_erase(const struct termios *t, char ch,
```
Comment (AntonKozlov) [inline src/drivers/tty/termios_ops.c]: could be `static`?

File: src/drivers/tty/termios_ops.c [hunk 2]
```diff
@@ -135,33 +151,76 @@ int termios_handle_erase(const struct termios *t, char ch, int *raw_mode,
...
 			/* maintain it empty */
-			i_canon_ring->tail = i_canon_ring->head;
+			b->canon_ring->tail = b->canon_ring->head;
 		}
 	}
 
 	return got_data;
 }
 
+int termios_input(const struct termios *t, char ch, int *is_ready,
```
Comment (AntonKozlov) [inline src/drivers/tty/termios_ops.c]: `termios_input` now returns bool `got_data`. You could pack `is_ready` and `got_data` in one int result, calculated from ORing values like 
```
#define TIO_RES_GOT_DATA (1 << 0)
#define TIO_RES_GOT_ECHO (1 << 1)
```

**Comment by AntonKozlov** [comment]
@anton-bondarev I'd like to merge this, but it could introduce tty related bugs. I think, that we could see regressions in console with tish, telnetd and dropbear.
What other things could break because bugs in tty, beside named above?
And when will be convinient for you to check all of that working, so after the merge we could check that again?

**Comment by AntonKozlov** [comment]
Tested console on x86/qemu and telnet there, works. 
@anton-bondarev OK to merge?

**Comment by anton-bondarev** [comment]
LGTM

**Pull request merged**